File: data/code_pairs/fortran/2235.f90
Program m5
  Implicit None
  Call s()
Contains
  Subroutine s
    Interface
      Function f()
        Real :: f
      End Function
      Function g()
        Real :: g
      End Function
    End Interface
    Print *, g()
    Print *, f()
    Print *, g()
  End Subroutine s

  Real Function f()
    f = 1.0
  End Function

  Real Function g()
    g = 2.0
  End Function
End Program m5

File: data/code_pairs/cpp/2235.cpp
#include <iostream>
#include <functional>

// Define the functions f and g
double f() {
    return 1;
}

double g() {
    return 2;
}

// Define the subroutine s
void s() {
    std::function<double()> p;

    // Assign p to point to function g and call it
    p = g;
    std::cout << p() << std::endl;

    // Assign p to point to function f and call it
    p = f;
    std::cout << p() << std::endl;

    // Assign p back to g and call it again
    p = g;
    std::cout << p() << std::endl;
}

int main() {
    s();
    return 0;
}